src/bijectors/named_bijector.jl

```diff
@@ -0,0 +1,231 @@
+
+forward(b::AbstractNamedBijector, x) = (rv = b(x), logabsdetjac = logabsdetjac(b, x))
+
+#######################
+### `NamedBijector` ###
+#######################
+"""
+    NamedBijector <: AbstractNamedBijector
+
+Wraps a `NamedTuple` of key -> `Bijector` pairs, implementing evaluation, inversion, etc.
+
+# Examples
+```julia-repl
+julia> using Bijectors: NamedBijector, Scale, Exp
+
+julia> b = NamedBijector((a = Scale(2.0), b = Exp()));
+
+julia> x = (a = 1., b = 0., c = 42.);
+
+julia> b(x)
+(a = 2.0, b = 1.0, c = 42.0)
+
+julia> (a = 2 * x.a, b = exp(x.b), c = x.c)
+(a = 2.0, b = 1.0, c = 42.0)
+```
+"""
+struct NamedBijector{names, Bs<:NamedTuple{names}} <: AbstractNamedBijector
+    bs::Bs
+end
+
+names_to_bijectors(b::NamedBijector) = b.bs
+
+@generated function (b::NamedBijector{names})(x::NamedTuple) where {names}
+    return :(merge(x, ($([:($n = b.bs.$n(x.$n)) for n in names]...), )))
+end
+
+@generated function Base.inv(b::NamedBijector{names}) where {names}
+    return :(NamedBijector(($([:($n = inv(b.bs.$n)) for n in names]...), )))
+end
+
+@generated function logabsdetjac(b::NamedBijector{names}, x::NamedTuple) where {names}
+    exprs = [:(logabsdetjac(b.bs.$n, x.$n)) for n in names]
+    return :(+($(exprs...)))
+end
+
+
+######################
+### `NamedInverse` ###
+######################
+"""
+    NamedInverse <: AbstractNamedBijector
+
+Represents the inverse of a `AbstractNamedBijector`, similarily to `Inverse` for `Bijector`.
+
+See also: [`Inverse`](@ref)
+"""
+struct NamedInverse{B<:AbstractNamedBijector} <: AbstractNamedBijector
+    orig::B
+end
+Base.inv(nb::AbstractNamedBijector) = NamedInverse(nb)
+Base.inv(ni::NamedInverse) = ni.orig
+
+logabsdetjac(ni::NamedInverse, y::NamedTuple) = -logabsdetjac(inv(ni), ni(y))
+
+##########################
+### `NamedComposition` ###
+##########################
+"""
+    NamedComposition <: AbstractNamedBijector
+
+Wraps a tuple of array of `AbstractNamedBijector` and implements their composition.
+
+This is very similar to `Composed` for `Bijector`, with the exception that we do not require
+the inputs to have the same "dimension", which in this case refers to the *symbols* for the
+`NamedTuple` that this takes as input.
+
+See also: [`Composed`](@ref)
+"""
+struct NamedComposition{Bs} <: AbstractNamedBijector
+    bs::Bs
+end
+
+# Essentially just copy-paste from impl of composition for 'standard' bijectors,
+# with minor changes here and there.
+composel(bs::AbstractNamedBijector...) = NamedComposition(bs)
+composer(bs::AbstractNamedBijector...) = NamedComposition(reverse(bs))
+∘(b1::AbstractNamedBijector, b2::AbstractNamedBijector) = composel(b2, b1)
+
+inv(ct::NamedComposition) = NamedComposition(reverse(map(inv, ct.bs)))
+
+function (cb::NamedComposition{<:AbstractArray{<:AbstractNamedBijector}})(x)
+    @assert length(cb.bs) > 0
+    res = cb.bs[1](x)
+    for b ∈ Base.Iterators.drop(cb.bs, 1)
+        res = b(res)
+    end
+
+    return res
+end
```

Comment:
Isn't this just
```suggestion
(cb::NamedComposition{<:AbstractArray{<:AbstractNamedBijector}})(x) = foldl(|>, cb.bs; init=x)
```
?

Reply:
Type-stability issue if output is different type than input, e.g. if we're using `TrackedArray` for parameters of the bijectors but not the input.

Reply:
Btw, this is why this particular impl was introduced for `Composition`, and so I used the same code for `NamedComposition`.

Reply:
If there are type stability problems with TrackedArray, you get them for sure with the current implementation here as well, don't you? (Below you avoid the issue by building one long expression)

Reply:
Only if the first bijector does not have tracked but later bijectors do, so I'd argue that the current impl covers more cases than the above, no?

Reply:
Yes, I guess it should be fine since the first call is handled separately. The repeated assignment of `res` could still be problematic depending on the output of the different bijectors, but I don't know if/how this could be fixed :shrug: